File: src/main/java/com/pestphp/pest/PestIcons.java
package com.pestphp.pest;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.util.IconLoader;
import com.intellij.ui.LayeredIcon;
import com.jetbrains.php.PhpIcons;

import javax.swing.*;

public class PestIcons {
    public static final Icon LOGO = IconLoader.getIcon("/logo.svg");
    public static final Icon RUN_SINGLE_TEST = new LayeredIcon(LOGO, AllIcons.RunConfigurations.TestState.Run);
    public static final Icon CONFIG = IconLoader.getIcon("/logo.svg");
}

File: src/main/kotlin/com/pestphp/pest/PestIcons.kt
package com.pestphp.pest

import com.intellij.icons.AllIcons
import com.intellij.ide.ui.LafManager
import com.intellij.ide.ui.laf.darcula.DarculaLookAndFeelInfo
import com.intellij.openapi.util.IconLoader

object PestIcons {
    val LOGO = IconLoader.getIcon("/logo@${getThemeString()}.svg", PestIcons.javaClass)
    val RUN_FILE_TEST = IconLoader.getIcon("/run@${getThemeString()}.svg", PestIcons.javaClass)
    val CONFIG = IconLoader.getIcon("/config@${getThemeString()}.svg", PestIcons.javaClass)
    val FILE = IconLoader.getIcon("/file@${getThemeString()}.svg", PestIcons.javaClass)
    val DATASET_FILE = IconLoader.getIcon("/dataset@${getThemeString()}.svg", PestIcons.javaClass)
    val SNAPSHOT_ICON = AllIcons.Nodes.DataSchema

    private fun isDarkMode() = LafManager.getInstance().currentLookAndFeel == DarculaLookAndFeelInfo()
    private fun getThemeString(): String {
        if (isDarkMode()) {
            return "dark"
        }
        return "light"
    }
}
